Sources/SwiftProtobufCore/TextFormatEncodingVisitor.swift

```diff
@@ -248,6 +248,10 @@ internal struct TextFormatEncodingVisitor: Visitor {
   }
 
   mutating func visitSingularBytesField(value: Data, fieldNumber: Int) throws {
+      try visitSingularBytesField(value: Array(value), fieldNumber: fieldNumber)
+  }
+
+  mutating func visitSingularBytesField(value: [UInt8], fieldNumber: Int) throws {
```

Comment:
I guess at some point this sorta change need to happen to the base Visitor?  Not sure what this will end up for the Compat module and api in general.

Reply:
Yes, the idea is to actually change the base visitor to be generic on a `SwiftProtobufContiguousBytes`-conforming type (i.e. `mutating func visitSingularBytesField<Bytes: SwiftProtobufContiguousBytes>(value: Bytes, fieldNumber: Int) throws`), so the Compat module wouldn't need to redefine the `Visitor` protocol or do anything special, since it will already be conforming `Data`. Was planning on making that change in the next PR to try to split the changes a bit.

Reply:
We should watch the performance impact of this.   Invoking generics across a module boundary is tricky to get right:  If you're not careful, you'll end up either ruining performance (because of meta type lookups on each call) or bloating code size (due to excessive inlining).  It might make sense to keep one or two statically-typed variations (e.g., `[UInt8]`) in addition to the generic version so that suitably-typed users can avoid that.

Reply:
I agree, I'll take special care and look into it when making that change. You think the changes so far are good enough though to go ahead with this PR?